File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {

        
        long[] answer = new long[2001];
        answer[0] = 1;
        answer[1] = 2;
        
        for(int i=2;i<n;i++){
            answer[i] = (answer[i-1] + answer[i-2]) % 1234567;
        }
        
        return answer[n-1];
    }
    
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <bits/stdc++.h>
using namespace std;

long long answer;




long long solution(int n) {
    
    if (n == 1) return 1;
    
    vector<int> v(n + 1, 0);
    
    v[1] = 1;
    v[2] = 2;
    
    for (int i=3;i<=n;i++) {
        v[i] = (v[i-2] + v[i-1]) % 1234567;
    }

    return v[n];
}